qemu/tests/memory_leak_after_nichotplug.py

```diff
@@ -0,0 +1,90 @@
+import logging
+import time
+
+from virttest import error_context
+from virttest import utils_net
+from virttest import utils_misc
+from virttest import utils_test
+from virttest import env_process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Memory leak check after nic hotplug/hotunplug
+    1) Boot a guest
+    2) Check free memory
+    3) Hotplug nic 100 times(windows) or add 300 vlan(linux)
+    4) Hotunplug nic 100 times(windows) or del 300 vlan(linux)
+    5) Check free memory again
+
+    :param test: kvm test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    params["start_vm"] = "yes"
+    env_process.preprocess_vm(test, params, env, params["main_vm"])
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    timeout = int(params.get("login_timeout", 360))
+    session = vm.wait_for_serial_login(timeout=timeout)
+    os_type = params.get("os_type")
+
+    free_mem_before_nichotplug = utils_misc.get_free_mem(session,
+                                                         os_type)
+    logging.info("Guest free memory before nic hotplug: %d",
+                 free_mem_before_nichotplug)
+
+    if os_type == "windows" or params.get("hotplug_unplug_test", "no") == "yes":
```

Comment:
Does not need `hotplug_unplug_test` either.

Reply:
@vivianQizhu Updated.